### PR TITLE
Add nginx static files serving config

### PR DESCRIPTION
**Pull request by ghost**
Currently, static files (*.png, *.js, etc) are served through Django, which is not ideal.

This PR changes configuration files so that NGINX will instead serve the static content.

This change is tied to https://github.com/open-craft/opencraft/pull/211

### CODE REVIEW

**Comment by snowcrshd** [comment]
@bradenmacdonald when you have the time, could you take a look at this PR and

**Comment by bradenmacdonald** [comment]
Looking good so far!

**Comment by ghost** [comment]
Added some commits after testing nginx config out.
I'll squash all before merge.

File: templates/opencraft-nginx.j2
Comment (smarnach) [inline templates/opencraft-nginx.j2]: What's the point of appending slashes for static assets?  Indexes are turned off, so I'm not sure what a path ending in a slash would do.  (I'm just curious what the use case is.  I'm not asking for this to be changed.)

File: defaults/main.yml
Comment (smarnach) [inline defaults/main.yml]: In some cases we use uppercase variables for defaults that are supposed to be changed by users of the role, and lower case variables for values that should not be overridden.  I'm not sure we are following this policy in this repo, but for consistency with the other directory names we should use a lower-case name here.

The static file directory is hard-coded in Ocim, so it definitely can't be changed by users in a meaningful way.

File: templates/opencraft-nginx.j2
Comment (smarnach) [inline templates/opencraft-nginx.j2]: Might be worth including `text/html` and `image/svg+xml` here.  (Angular templates are loaded as static assets, and they probably have MIME type `text/html`.)

**Review by smarnach — CHANGES_REQUESTED** [review]
@snowcrshd Thanks for working on this!  I had a look at the code and added a few questions and comments.

File: templates/opencraft-nginx.j2
Comment (ghost) [reply templates/opencraft-nginx.j2]: @smarnach oh, I forgot to get rid of this.

I was using `autoindex on;` before.
Thanks for spotting that!

File: templates/opencraft-nginx.j2
Comment (ghost) [reply templates/opencraft-nginx.j2]: Done!

File: defaults/main.yml
Comment (ghost) [reply defaults/main.yml]: @smarnach this is good to know.
I was not sure what was the guideline here, as I had seen both lower and uppercase variables.

I'll change this to lowercase and update the PR.

**Comment by ghost** [comment]
@smarnach thanks for the review! I hadn't noticed the points you mentioned  😅 

Just fixed them and updated the PR

**Comment by smarnach** [comment]
:+1: Looks good to me.  You can squash your commits now, and I will merge it.

Can you please update the PR description?  It's helpful when the description explains what the PR does in case someone comes back here.  Just a sentence or two will do.

- [x] I tested this: I deployed this to the Ocim stage instance, and it works great.
- [x] I read through the code
- [ ] ~~I checked for accessibility issues~~
- [ ] ~~Includes documentation~~

**Review by smarnach — APPROVED** [review]

**Comment by ghost** [comment]
@smarnach commits squashed and description updated!

**Comment by smarnach** [comment]
@snowcrshd Thanks!  I will merge this once we have a green build for https://github.com/open-craft/opencraft/pull/211

**Comment by ghost** [comment]
@smarnach great!

**Comment by ghost** [comment]
@bradenmacdonald does this PR too need CircleCI to pass for us to merge it?

**Comment by bradenmacdonald** [comment]
@snowcrshd No, we don't have CI enabled for this repo.

**Pull request merged**
